spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/MojoHarnessTest.java

```diff
@@ -25,78 +25,72 @@
 import java.util.Properties;
 import java.util.stream.Collectors;
 
+import org.apache.maven.execution.DefaultMavenExecutionRequest;
+import org.apache.maven.execution.MavenExecutionRequest;
 import org.apache.maven.model.Dependency;
 import org.apache.maven.model.Model;
 import org.apache.maven.model.Parent;
 import org.apache.maven.model.Plugin;
 import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
-import org.apache.maven.plugin.testing.MojoRule;
+import org.apache.maven.plugin.testing.AbstractMojoTestCase;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuilder;
+import org.apache.maven.project.ProjectBuildingRequest;
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * @author Christian Tzolov
  */
-public class MojoHarnessTest {
+class MojoHarnessTest extends AbstractMojoTestCase {
 
-	@Rule
-	public TemporaryFolder projectHome = new TemporaryFolder();
-
-	@Rule
-	public MojoRule mojoRule = new MojoRule();
+	@BeforeEach
+	void prepareForTest() throws Exception {
+		super.setUp();
+	}
 
 	@Test
-	public void testSomething() throws Exception {
-
-		File pomRoot = new File("src/test/resources/unit/http-source-apps/");
-
-		SpringCloudStreamAppGeneratorMojo myMojo = (SpringCloudStreamAppGeneratorMojo)
-				mojoRule.lookupConfiguredMojo(pomRoot, "generate-app");
-
+	void httpSourceAppsGeneration() throws Exception {
+		File pomRoot = new File("src/test/resources/unit/http-source-apps");
+		MavenProject project = readMavenProject(pomRoot);
+		SpringCloudStreamAppGeneratorMojo myMojo = (SpringCloudStreamAppGeneratorMojo) super.lookupConfiguredMojo(project, "generate-app");
```

Comment:
There is no Junit5 replacement for the Mojo Junit5 rule. **HOWEVER**, the `AbstractMojoTestCase` has what we need but we have to jump through some hoops in `readMavenProject` to get an actual project object to then pass into `super.lookupConfiguredMojo`.